docker/docker.hcl

```diff
@@ -0,0 +1,23 @@
+group "default" {
+  targets = ["graphql-mesh"]
+}
+
+variable "PWD" {
+  default = "."
+}
+
+variable "context" {
+  default = "${PWD}/docker"
+}
+
+target "graphql-mesh" {
```

Comment:
I don't think this should be named just `mesh`. If this image is intended for a specific variant for Hive <> Mesh integration, then it should be named like that. 

`graphql-mesh-hive`  ? @n1ru4l

Reply:
this image is supposed to run generally, and can be flipped into a hive integration with the right environment variables set, so it should probably be just `mesh` or `graphql-mesh` since it's not hive specific